### PR TITLE
Alibaba: Update tag and API types to mirror other providers

### PR DESCRIPTION
**Pull request by kwoodson**
Work summary:
- Update vendor dependent libraries to no longer depend on machine-api-operator's machine api
- Updating the tag types used to search for provider objects
- Remove DNS entry for `*.apps`. This is created by the ingress-operator.

### CODE REVIEW

**Comment by staebler** [comment]
I am going to assume that this is a WIP PR since it includes numerous disjoint changes and a vendor of a personal repo.

/retitle [WIP] Alibaba: Update tag types to mirror other providers

**Comment by kwoodson** [comment]
@staebler Thanks, I should have tidied this up. I'm waiting for the mapi to update and then this one is eligible.

**Comment by kwoodson** [comment]
Seems like cluster was still installing but test timed out (ovirt)?
```
fail [github.com/onsi/ginkgo@v4.7.0-origin.0+incompatible/internal/leafnodes/runner.go:113]: Nov 12 22:31:42.482: ClusterVersion Progressing=True: : Working towards 4.10.0-0.ci.test-2021-11-12-220056-ci-op-s4nky6rq-latest: 705 of 773 done (91% complete)
```

File: pkg/asset/machines/master.go [hunk 1]
```diff
@@ -544,7 +542,7 @@ func (m *Master) Machines() ([]machineapi.Machine, error) {
 	libvirtapi.AddToScheme(scheme)
 	openstackapi.AddToScheme(scheme)
 	ovirtproviderapi.AddToScheme(scheme)
-	vsphereapi.AddToScheme(scheme)
+	machineapi.AddToScheme(scheme)
```
Comment (staebler) [inline pkg/asset/machines/master.go:545]: See https://github.com/openshift/installer/pull/5350#pullrequestreview-806725545 for why this won't work.

**Review by staebler — COMMENTED** [review]
The last commit seems unrelated to tagging.

The other commits could use some elaboration in their commits messages. For example, "update and simplify tag types" does not tell me anything about how or why the tag types are being updated or simplified.

File: pkg/asset/machines/master.go (hunk 1 above)
Comment (kwoodson) [reply pkg/asset/machines/master.go:545]: @staebler Since there are multiple PRs attempting to solve this issue, what do you recommend?

**Comment by kwoodson** [comment]
@staebler I have updated the commit message and removed the last commit and created a new PR #5396

**Comment by staebler** [comment]
> @staebler I have updated the commit message and removed the last commit and created a new PR #5396

@kirankt is working on it. I recommend waiting for his fix. Is this something that is blocking this PR?

File: pkg/asset/machines/master.go (hunk 1 above)
Comment (staebler) [reply pkg/asset/machines/master.go:545]: @kirankt is working on it. I recommend waiting for his fix. Is this something that is blocking this PR?

Comment (kwoodson) [reply pkg/asset/machines/master.go:545]: @staebler The issue here is that the cluster-api-provider-alibaba requires a newer version of the machine-api-operator. The vsphere provider requires an older version. Those conflict based upon the machine libraries moving to openshift/api. There might be some go.mod pinning to work around this but I wanted to provide the correct fix by cleaning up the installer dependencies and moving the requirement to openshift/api. I will wait to see what @kirankt comes up with unless you have a suggestion.

Comment (kirankt) [reply pkg/asset/machines/master.go:545]: I don't have a good solution. The recent vendoring brought back MAO probably because of cap-alibaba deps.

Comment (kwoodson) [reply pkg/asset/machines/master.go:545]: @kirankt The issue here isn't cap-Alibaba. The dependency for mao in cap-Alibaba is recent. The issue is we need to clean up the vsphere libraries. The vsphere and openstack caps depend on the older version. I created merged PRs in both of those repositories to update the mao dependency.

Comment (kwoodson) [reply pkg/asset/machines/master.go:545]: @kirankt I checked out your PR. It appears that you are adding the scheme and addressing @staebler's feedback. I'll await  your changes and will rebase when yours completes. Thanks

**Comment by kwoodson** [comment]
Awaiting #5350 and then will clean this PR up.

File: pkg/types/alibabacloud/machinepool.go [hunk 2]
```diff
@@ -29,18 +33,10 @@ type MachinePool struct {
...
-	SystemDiskCategory DiskCategory `json:"systemDiskCategory,omitempty"`
-
-	// SystemDiskSize defines the size of the system disk in gibibytes (GiB).
+	// SystemDiskInfo defines the category of the system disk.
 	//
-	// +kubebuilder:validation:Type=integer
-	// +kubebuilder:validation:Minimum=120
 	// +optional
-	SystemDiskSize int `json:"systemDiskSize,omitempty"`
+	SystemDisk machinev1.SystemDiskProperties `json:"systemDisk,omitempty"`
```
Comment (staebler) [inline pkg/types/alibabacloud/machinepool.go:39]: Is this change intentional? There is no mention of it in the commit description. Generally, it is discouraged to use a type defined elsewhere in the install config. Doing so makes it much more difficult to ensure the API stability of the install config.

File: go.mod [hunk 3]
```diff
@@ -104,18 +103,18 @@ require (
...
-	google.golang.org/grpc v1.32.0
-	gopkg.in/ini.v1 v1.61.0
+	google.golang.org/api v0.44.0
+	google.golang.org/genproto v0.0.0-20210602131652-f16073e35f0c
+	google.golang.org/grpc v1.38.0
+	gopkg.in/ini.v1 v1.62.0
 	gopkg.in/yaml.v2 v2.4.0
 	k8s.io/api v0.22.2
-	k8s.io/apiextensions-apiserver v0.21.0-rc.0
+	k8s.io/apiextensions-apiserver v0.22.0-rc.0
```
Comment (staebler) [inline go.mod:112]: If we are changing this anyway, we may as well change it to v0.22.2 so that it matches the version of k8s.io/api that we are using.
```suggestion
	k8s.io/apiextensions-apiserver v0.22.2
```

File: pkg/types/alibabacloud/machinepool.go (hunk 2 above)
Comment (kwoodson) [reply pkg/types/alibabacloud/machinepool.go]: It was intentional to align with the new API types. I have gone back and replaced it with what was previously there from your comments.

**Comment by kwoodson** [comment]
@staebler Please review. I've updated according to your comments and was able to install successfully.
```
INFO Install complete!                            
INFO To access the cluster as the system:admin user when using 'oc', run 'export KUBECONFIG=/home/kwoodson/tmp/alibaba/cluster/auth/kubeconfig' 
INFO Access the OpenShift web-console here: https://console-openshift-console.apps.test.alicloud-dev.devcluster.openshift.com 
INFO Login to the console with user: "kubeadmin", and password: "xxxxxxxxx" 
DEBUG Time elapsed per stage:                      
DEBUG            cluster: 2m14s                    
DEBUG          bootstrap: 1m1s                     
DEBUG Bootstrap Complete: 19m56s                   
DEBUG                API: 3m12s                    
DEBUG  Cluster Operators: 30m16s                   
INFO Time elapsed: 53m33s                         

```

File: pkg/asset/machines/alibabacloud/machines.go [hunk 4]
```diff
@@ -98,21 +98,21 @@ func provider(clusterID string,
...
+			Category: string(mpool.SystemDiskCategory),
+			Size:     mpool.SystemDiskSize,
+		},
+		RegionID:          platform.Region,
+		ResourceGroupID:   platform.ResourceGroupID,
+		ZoneID:            az,
+		UserDataSecret:    &corev1.LocalObjectReference{Name: userDataSecret},
+		CredentialsSecret: &corev1.LocalObjectReference{Name: "alibabacloud-credentials"},
+		Tags:              tags,
+		// InstanceName:      fmt.Sprintf("%s-%s", clusterID, role),
```
Comment (staebler) [inline pkg/asset/machines/alibabacloud/machines.go:113]: Can we remove this commented-out line?

File: pkg/types/alibabacloud/machinepool.go [hunk 5]
```diff
@@ -39,7 +39,7 @@ type MachinePool struct {
 	//
 	// +kubebuilder:validation:Type=integer
 	// +kubebuilder:validation:Minimum=120
-	// +optional
+	// +optional		￼	// +optional
```
Comment (staebler) [inline pkg/types/alibabacloud/machinepool.go:42]: Leftover changes.

**Review by staebler — COMMENTED** [review]
Looks good save for a couple minor non-functional changes.

File: pkg/asset/machines/alibabacloud/machines.go (hunk 4 above)
Comment (kwoodson) [reply pkg/asset/machines/alibabacloud/machines.go:113]: Removed.

File: pkg/types/alibabacloud/machinepool.go (hunk 5 above)
Comment (kwoodson) [reply pkg/types/alibabacloud/machinepool.go]: Fixed.

**Comment by kwoodson** [comment]
Updated vendor to resolve conflicts.

**Comment by staebler** [comment]
/lgtm cancel
I think you have merge conflicts now, probably against https://github.com/openshift/installer/pull/5402.

File: go.mod [hunk 6]
```diff
@@ -52,25 +51,32 @@ require (
...
-	github.com/openshift/machine-api-operator v0.2.1-0.20210505133115-b7ef098180db
+	github.com/openshift/cluster-api-provider-ovirt v0.1.1-0.20211111151530-06177b773958
+	github.com/openshift/library-go v0.0.0-20210811133500-5e31383de2a7
 	github.com/openshift/machine-config-operator v0.0.0
 	github.com/ovirt/go-ovirt v0.0.0-20210308100159-ac0bcbc88d7c
 	github.com/ovirt/terraform-provider-ovirt v0.99.1-0.20211019085223-db1ac552ec57
 	github.com/pborman/uuid v1.2.0
 	github.com/pkg/errors v0.9.1
 	github.com/pkg/sftp v1.10.1
+<<<<<<< HEAD
```
Comment (staebler) [inline go.mod:68]: Unresolved conflicts.

Comment (kwoodson) [reply go.mod]: I noticed this moved to go 1.17.  Updated.

Thanks!

**Comment by kwoodson** [comment]
@staebler I fixed the rebase issues from the previous go 1.17 update. Please TAL

**Comment by staebler** [comment]
This looks good to me. I'd like to see a successful openstack and ovirt e2e job. The e2e-ovirt history looks pretty bad, unrelated to this PR. We should be able to get a successful openstack job (note that the openshift-kuryr job was successful).

/test e2e-openstack
/test e2e-ovirt

**Comment by kwoodson** [comment]
@staebler Looks like e2e-{ovirt,openstack} are now passing. Mind bumping this along?

**Comment by zhouhao3** [comment]
Hi @kwoodson Is there any special use for [specifying the version of gophercloud](https://github.com/openshift/installer/pull/5381/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R376) in gomod? I noticed that you [updated the version ](https://github.com/openshift/installer/pull/5381/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R36)of gophercloud and then specified the version of gophercloud below.
